pkg/statistics/table.go

```diff
@@ -662,6 +663,85 @@ func (t *Table) ShallowCopy() *Table {
 	return nt
 }
 
+// CopyForColumnUpdate creates a copy optimized for column updates.
+func (t *Table) CopyForColumnUpdate() *Table {
+	newColumns := make(map[int64]*Column, len(t.columns))
+	for id, col := range t.columns {
+		newColumns[id] = col
+	}
+
+	newHistColl := HistColl{
+		PhysicalID:    t.PhysicalID,
+		RealtimeCount: t.RealtimeCount,
+		columns:       newColumns,
+		indices:       t.indices,
+		Pseudo:        t.Pseudo,
+		ModifyCount:   t.ModifyCount,
+		StatsVer:      t.StatsVer,
+	}
+	nt := &Table{
+		HistColl:              newHistColl,
+		Version:               t.Version,
+		TblInfoUpdateTS:       t.TblInfoUpdateTS,
+		ExtendedStats:         t.ExtendedStats,
+		ColAndIdxExistenceMap: t.ColAndIdxExistenceMap.Clone(),
+		LastAnalyzeVersion:    t.LastAnalyzeVersion,
+		LastStatsHistVersion:  t.LastStatsHistVersion,
+	}
+	return nt
+}
+
+// CopyForIndexUpdate creates a copy optimized for index updates.
+func (t *Table) CopyForIndexUpdate() *Table {
+	newIndices := make(map[int64]*Index, len(t.indices))
+	for id, idx := range t.indices {
+		newIndices[id] = idx
```

Comment:
I noticed in deep copy, we copy the idx object as well, such as 
`newHistColl.indices[id] = idx.Copy()`

And I wonder if that's necessary for all our use cases since it seems our updates are adding/removing new col instead of changing the current ones, I wonder for the other copy cases, can we do shallow copy instead?